src/app/fdctl/run/run.c

```diff
@@ -61,145 +74,35 @@ install_tile_signals( void ) {
     FD_LOG_ERR(( "sigaction(SIGINT) failed (%i-%s)", errno, fd_io_strerror( errno ) ));
 }
 
-static int
-getpid1( void ) {
-  char pid[ 12 ] = {0};
-  long count = readlink( "/proc/self", pid, sizeof(pid) );
-  if( FD_UNLIKELY( count < 0 ) ) FD_LOG_ERR(( "readlink(/proc/self) failed (%i-%s)", errno, fd_io_strerror( errno ) ));
-  if( FD_UNLIKELY( (ulong)count >= sizeof(pid) ) ) FD_LOG_ERR(( "readlink(/proc/self) returned truncated pid" ));
-  char * endptr;
-  ulong result = strtoul( pid, &endptr, 10 );
-  if( FD_UNLIKELY( *endptr != '\0' || result > INT_MAX  ) ) FD_LOG_ERR(( "strtoul(/proc/self) returned invalid pid" ));
-
-  return (int)result;
-}
-
-int
-tile_main( void * _args ) {
-  tile_main_args_t * args = _args;
-
-  fd_topo_tile_t * tile = args->tile;
-
-  fd_log_private_tid_set( tile->id );
-  fd_log_thread_set( fd_topo_tile_kind_str( tile->kind ) );
-
-  int pid = getpid1(); /* need to read /proc since we are in a PID namespace now */
-  fd_log_private_group_id_set( (ulong)pid );
-  FD_LOG_NOTICE(( "booting tile %s(%lu) pid(%d)", fd_topo_tile_kind_str( tile->kind ), tile->kind_id, pid ));
-
-  install_tile_signals();
-
-  /* calling fd_tempo_tick_per_ns requires nanosleep, it is cached with
-     a FD_ONCE.  We do this for all tiles before sandboxing so that we
-     don't need to allow the nanosleep syscall. */
-  fd_tempo_tick_per_ns( NULL );
-
-  /* preload shared memory before sandboxing, so it is already mapped */
-  fd_topo_join_tile_workspaces( args->config->name, &args->config->topo, tile );
-
-  fd_tile_config_t * config = fd_topo_tile_to_config( tile );
-
-  void * scratch_mem   = NULL;
-  if( FD_LIKELY( config->scratch_align ) ) {
-    scratch_mem = (uchar*)args->config->topo.workspaces[ tile->wksp_id ].wksp + tile->user_mem_offset;
-    if( FD_UNLIKELY( !fd_ulong_is_aligned( (ulong)scratch_mem, config->scratch_align() ) ) )
-      FD_LOG_ERR(( "scratch_mem is not aligned to %lu", config->scratch_align() ));
-  }
+extern int * fd_log_private_shared_lock;
```

Comment:
I think you can get rid of this line now, right? Seems unused. Same for in main1.c